src/javascript/request-composer/components/request-composer.js

```diff
@@ -0,0 +1,614 @@
+        </Tabs>
+
+        <h3 className="H3--underline">Set the query parameters</h3>
+
+        <form onSubmit={this.handleSubmit}>
+
+          <div className={requiredFormControlClass}>
+            <label className="FormControl-label">viewId</label>
+            <div className="FormControl-body">
+              <div className="FlexLine">
+                <input
+                  className="FormField FormFieldCombo-field"
+                  name="viewId"
+                  value={params.viewId}
+                  onChange={this.handleParamChange} />
+                <HelpIconLink
+                  url={REFERENCE_URL}
+                  name="ReportRequest.FIELDS.view_id" />
+              </div>
+            </div>
+          </div>
+
+          {settings.requestType != 'COHORT' ? (
+          <div className={requiredFormControlClass}>
+            <label className="FormControl-label">startDate</label>
+            <div className="FormControl-body">
+              <div className="FlexLine">
+                <Datepicker
+                  name="startDate"
+                  value={params['startDate']}
+                  onChange={this.handleParamChange} />
+                <HelpIconLink
+                  url={REFERENCE_URL}
+                  name="DateRange.FIELDS.start_date" />
+              </div>
+            </div>
+          </div>
+          ) :
+          null}
+
+          {settings.requestType != 'COHORT' ? (
+          <div className={requiredFormControlClass}>
+            <label className="FormControl-label">endDate</label>
+            <div className="FormControl-body">
+              <div className="FlexLine">
+                <Datepicker
+                  name="endDate"
+                  value={params['endDate']}
+                  onChange={this.handleParamChange} />
+                <HelpIconLink
+                  url={REFERENCE_URL}
+                  name="DateRange.FIELDS.start_date" />
+              </div>
+            </div>
+          </div>
+          ) :
+          null}
+
+          {settings.requestType != 'COHORT' ? (
+          <div className={requiredFormControlClass}>
+            <label className="FormControl-label">metrics</label>
+            <div className="FormControl-body">
+              <div className="FlexLine">
+                <Select2MultiSuggest
+                  name="metrics"
+                  value={params.metrics}
+                  tags={select2Options.metrics}
+                  onChange={this.handleParamChange}
+                  maximumSelectionSize={maximumSelectionSize} />
+                <HelpIconLink
+                  url={REFERENCE_URL}
+                  name="ReportRequest.FIELDS.metrics" />
+              </div>
+            </div>
+          </div>
+          ) :
+          null}
+
+
+          {settings.requestType == 'HISTOGRAM' ? (
+          <div className={formControlClass}>
+            <label className="FormControl-label">Histogram dimension</label>
```

Comment:
A lot of these parameters aren't shown as being required, and if I try to make requests without them being present, I get JS errors instead of UI errors. Can you make sure all required fields are visually indicated as such via the `requiredFormControlClass` `className` prop.